pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -35,45 +150,37 @@ def test_upload(self):
            ``--skip-existing`` flag during the upload. Verify that Pulp skips
            the upload.
         """
-        if selectors.bug_is_untestable(1806, config.get_config().version):
+        if selectors.bug_is_untestable(1806, self.cfg.version):
             self.skipTest('https://pulp.plan.io/issues/1806')
 
         # Create a repository
-        client = cli.Client(config.get_config())
         repo_id = utils.uuid4()
-        client.run(
-            'pulp-admin rpm repo create --repo-id {}'.format(repo_id).split()
-        )
-        self.addCleanup(
-            client.run,
-            'pulp-admin rpm repo delete --repo-id {}'.format(repo_id).split()
-        )
-
-        # Create a temporary directory, and download a DRPM file into it
-        temp_dir = client.run('mktemp --directory'.split()).stdout.strip()
-        self.addCleanup(client.run, 'rm -rf {}'.format(temp_dir).split())
-        drpm_file = os.path.join(temp_dir, os.path.split(DRPM)[-1])
-        client.run(
-            'curl -o {} {}'.format(drpm_file, DRPM_UNSIGNED_URL).split()
-        )
+        self.create_repo('rpm', repo_id)
+        drpm_file_path = self.temp_download(DRPM_UNSIGNED_URL, DRPM)
 
         # Upload the DRPM into the repository. Don't use subTest, as if this
         # test fails, the following one is invalid anyway.
-        client.run(
-            'pulp-admin rpm repo uploads drpm --repo-id {} --file {}'
-            .format(repo_id, drpm_file).split()
+        self.upload_to_repo(
+            plugin='rpm',
+            filetype='drpm',
+            repo_id=repo_id,
+            filename=drpm_file_path,
         )
-        proc = client.run(
-            'pulp-admin rpm repo content drpm --repo-id {} --fields filename'
-            .format(repo_id).split()
+        self.check_field(
+            plugin='rpm',
+            filetype='drpm',
+            repo_id=repo_id,
+            expected=DRPM,
+            search_field='filename'
         )
-        self.assertEqual(proc.stdout.split('Filename:')[1].strip(), DRPM)
 
         # Upload the DRPM into the repository. Pass --skip-existing.
-        proc = client.run(
-            ('pulp-admin rpm repo uploads drpm --repo-id {} --file {} '
-             '--skip-existing')
-            .format(repo_id, drpm_file).split()
+        proc = self.upload_to_repo(
+            plugin='rpm',
+            filetype='drpm',
+            repo_id=repo_id,
+            filename=drpm_file_path,
+            args='--skip-existing',
```

Comment:
The word `args` isn't a keyword, but it's conventionally used to indicate a list of un-named function parameters:

```python
def print_args(*args):
    for arg in args:
        print(arg)
```

I think the design of the existing RPM CLI helper functions in `pulp_smash.tests.docker.cli.utils` may serve as a good starting point, design-wise.